src/client/info-client.ts

```diff
@@ -51,7 +51,7 @@ async function fetchEdgeServers(
   opts: CommonOptions = {}
 ): Promise<EdgeServers> {
   const { log = noOp, fetch = crossFetch } = opts
-  let error: Error = new Error('No info servers')
+  let error: unknown = new Error('No info servers')
```

Comment:
Makes sense. JavaScript is lame in the way that anything can be thrown and caught.

It'd be nice to abstract away managing this by having a catcher that will wrap other things thrown in an `Error`. There could be a clear protocol here for wrapping the types (e.g. strings are used as messages, objects are perhaps encoded with JSON or otherwise wrapped in a special error type that extends Error, etc). I have some APIs which could do this:

```js
trial<T>(tryFunc, catchFunc, catchFunc, ...)
```

The signature for each catch argument: `(err: Error)`. If a catch throws, `errorCause` could be used to chain errors so we know the causal chain in the trial.

Reply:
I think the rule should be, "clean what you catch", just as we do with reading stuff off of disk or network.